sigstore-public-deployment-spec.md

```diff
@@ -0,0 +1,172 @@
+
+## 3. Rekor
+
+Rekor implements a transparency service. There is a public good deployment of Rekor run by the [OpenSSF](https://openssf.org/) and contributing organizations at [https://rekor.sigstore.dev/](https://rekor.sigstore.dev/).
+
+### 3.1 Pluggable Types
+
+The transparency service has what is termed a ‘pluggable type’ system. A pluggable type, is a custom schema for entries stored in the transparency log. Schemas can be in multiple formats (json|yaml|xml).
+
+The current list of supported types can be found in the [Rekor project](https://github.com/sigstore/rekor/tree/main/pkg/types). Information about adding new pluggable types can be found in the [Rekor documentation.](https://docs.sigstore.dev/docs/logging/pluggable-types/)
+
+See the transparency service ([Spec: Rekor](https://docs.google.com/document/u/0/d/1NQUBSL9R64_vPxUEgVKGb0p81_7BVZ7PQuI078WFn-g/edit)) document for additional information.
+
+### 3.2 Transparency Log
+
+Rekor is backed by a transparency log, inspired by the one in Certificate Transparency ([RFC 6962](https://datatracker.ietf.org/doc/rfc6962/)). The primary deviations are described in [Spec: Rekor](https://docs.google.com/document/u/0/d/1NQUBSL9R64_vPxUEgVKGb0p81_7BVZ7PQuI078WFn-g/edit); however, like RFC 6962, the transparency service specification leaves specific parameter choices to implementers.
+
+* Base URL: [https://rekor.sigstore.dev/](https://rekor.sigstore.dev/)   
+* Hash Algorithm: SHA-256 ([RFC 6234](https://datatracker.ietf.org/doc/rfc6234/); OID 2.16.840.1.101.3.4.2.1)  
+* Signature Algorithm:  ECDSA (NIST P-256).  
+* Public Key: change over time  
+* Log ID: need an OID  
+* Maximum Merge Delay: Rekor only returns after the merge is complete  
+* Maximum Chain Length: 10  
+* STH Frequency Count: N/A  
+* Final STH: N/A
+
+### 3.3 Sharding
+
+The Certificate Transparency Log database used by Rekor currently shards every year at a minimum. Each shard is available but kept as a separate database instance.
+
+The sharding is an additive in a sense that old shards will be still available at:  `https://ctfe.sigstore.dev/<SHARD>`. For example, if sharding in 2022, the old shard is available at: [https://ctfe.sigstore.dev/2022/ct/v1/get-sth](https://ctfe.sigstore.dev/2022/ct/v1/get-sth)
+
+The convention for naming shards is that it will contain the year, followed by the instance. For example, the first shard of the year 2023 should be named 2023 and if other shards are created they will be called 2023-2, 2023-3, and so forth.
+
+### 3.4 Timestamp Authority
+
+Sigstore does not operate a timestamp authority at this time. We do include trusted timestamp authorities in Sigstore's TUF root. Signed timestamps can also be obtained through Rekor's SignedEntryTimestamps.
```

Comment:
We are going to remove timestamp authorities from the trust root.

We should significantly flesh this section out since it's not well discussed anywhere. This could have its own top level header, "Verifying short-lived certificates" and discuss sources of time (signed timestamps from TSAs or from Rekor) and the threat model of trusting each (distributed trust vs trusting Rekor)

Again, something to be changed in v2, separately out the timestamping portion from rekor.

Reply:
Tracking as issue #34